src/lib.rs

```diff
@@ -192,21 +185,44 @@ pub fn build(config: &Config) -> Result<(), Box<std::error::Error>> {
 
     document.data = Some(PrimaryData::Single(Box::new(krate)));
 
-    let serialized = serde_json::to_string(&document)?;
+    Ok(document)
+}
 
-    let mut file = File::create(json_path)?;
-    file.write_all(serialized.as_bytes())?;
+pub fn build(config: &Config, artifacts: &[&str]) -> Result<(), Box<std::error::Error>> {
```

Comment:
Actually, since `artifacts` is only used here, that's why I passed it in as opposed to keeping it in the `Config` struct. Otherwise it would be passed into all other subcommands. In the future we could turn it into a `BuildConfig` struct with build-specific options?